LPS-86406 AOP 2

Preston-Crary opened the pull request.

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

Preston-Crary commented:
Tests are passing. Changes from last time:
1. Fixed the bug with the DynamicDataSourceMethodInterceptor.
2. Made the order and default MethodInterceptors configurable in portal.properties.
3. Merged the FactoryProvider and the Factory interface and implementations.
4. Combined the Clusterable MethodInterceptorFactories.
5. Removed instead of deprecating code that has no way of working correctly after the change.

Preston-Crary commented:
Removed the packageinfos by not including the new packages in the bnd.
Renamed the helper to context.

Preston-Crary commented:
Made some more minor changes and included the example module usage.
